lib/full-site-editing/block-templates.php

```diff
@@ -241,31 +241,35 @@ function _gutenberg_build_template_result_from_file( $template_file, $template_t
 /**
  * Build a unified template object based a post Object.
  *
- * @param WP_Post $post Template post.
+ * @param WP_Post $post          Template post.
+ * @param string  $template_type wp_template or wp_template_part.
  *
  * @return WP_Block_Template|WP_Error Template.
  */
-function _gutenberg_build_template_result_from_post( $post ) {
-	$terms = get_the_terms( $post, 'wp_theme' );
-
-	if ( is_wp_error( $terms ) ) {
-		return $terms;
+function _gutenberg_build_template_result_from_post( $post, $template_type = 'wp_template' ) {
+	if ( $template_type !== $post->post_type ) {
+		return new WP_Error( 'template_wrong_post_type', __( 'An invalid post was provided for this template.', 'gutenberg' ) );
 	}
 
-	if ( ! $terms ) {
+	$ids    = get_theme_mod( $template_type, array() );
+	$active = in_array( $post->ID, $ids, true );
+
+	// Temporarily disable inactive access for 5.8 version.
+	if ( ! $active ) {
 		return new WP_Error( 'template_missing_theme', __( 'No theme is defined for this template.', 'gutenberg' ) );
 	}
 
-	$theme          = $terms[0]->name;
-	$has_theme_file = wp_get_theme()->get_stylesheet() === $theme &&
-		null !== _gutenberg_get_template_file( $post->post_type, $post->post_name );
+	$theme          = $active ? wp_get_theme()->get_stylesheet() : '';
+	$slug           = $active ? array_search( $post->ID, $ids, true ) : '';
+	$has_theme_file = $active &&
+		null !== _gutenberg_get_template_file( $post->post_type, $slug );
```

Comment:
Since we bail early `if ( ! active )`, we can simplify all the ternaries now:

```suggestion
	$theme          = wp_get_theme()->get_stylesheet();
	$slug           = array_search( $post->ID, $ids, true );
	$has_theme_file = null !== _gutenberg_get_template_file( $post->post_type, $slug );
```

I know that we might need to re-introduce it later, but I'm a firm believer in the YAGNI pattern and would thus only do that once we really do need it -- too many things can change until then 😄